translator/opencensus/oc_to_internal.go

```diff
@@ -95,34 +102,24 @@ func ocToInternal(td consumerdata.TraceData) data.TraceData {
 			// Add the span to the "combinedSpans". combinedSpans length is equal
 			// to combinedSpanCount. The loop above that calculates combinedSpanCount
 			// has exact same conditions as we have here in this loop.
-			destSpan := combinedSpans[i]
+			ocSpanToInternal(combinedSpans.Get(i), ocSpan)
 			i++
-
-			// Convert the span.
-			ocSpanToInternal(destSpan, ocSpan)
 		} else {
 			// This span has a different Resource and must be placed in a different
 			// ResourceSpans instance. Create a separate ResourceSpans item just for this span.
-			separateRS := ocSpanToResourceSpans(ocSpan, td.Node, ocSpan.Resource)
-
-			// Add the newly created ResourceSpans to the final list that we will return.
-			resourceSpanList = append(resourceSpanList, separateRS)
+			ocSpanToResourceSpans(ocSpan, td.Node, traceData.ResourceSpans().Get(j))
+			j++
 		}
 	}
 
-	return data.NewTraceData(resourceSpanList)
+	return traceData
 }
 
-func ocSpanToResourceSpans(ocSpan *octrace.Span, node *occommon.Node, resource *ocresource.Resource) *data.ResourceSpans {
-	destSpan := data.NewSpan()
-	ocSpanToInternal(destSpan, ocSpan)
-
-	// Create a separate ResourceSpans item just for this span.
-	return data.NewResourceSpans(
-		ocNodeResourceToInternal(node, resource),
-		[]*data.InstrumentationLibrarySpans{
-			data.NewInstrumentationLibrarySpans(data.NewInstrumentationLibrary(), []data.Span{destSpan})},
-	)
+func ocSpanToResourceSpans(ocSpan *octrace.Span, node *occommon.Node, out data.ResourceSpans) {
+	ocNodeResourceToInternal(node, ocSpan.Resource, out)
+	out.SetInstrumentationLibrarySpans(data.NewInstrumentationLibrarySpansSlice(1))
+	out.InstrumentationLibrarySpans().Get(0).SetSpans(data.NewSpanSlice(1))
```

Comment:
Would nice to use temp var for `out.InstrumentationLibrarySpans().Get(0)`.

Reply:
done.